packages/store/node-server-sdk-redis/src/RedisClientState.ts

```diff
@@ -0,0 +1,128 @@
+import { LDLogger } from '@launchdarkly/node-server-sdk';
+import { Redis } from 'ioredis';
+import LDRedisOptions from './LDRedisOptions';
+
+const DEFAULT_PREFIX = 'launchdarkly';
+
+/**
+ * Class for managing the state of a redis connection.
+ *
+ * Used for the redis persistent store as well as the redis big segment store.
+ *
+ * @internal
+ */
+export default class RedisClientState {
+  private connected: boolean = false;
+
+  private attempt: number = 0;
+
+  private initialConnection: boolean = true;
+
+  private client: Redis;
+
+  private owned: boolean;
+
+  private base_prefix: string;
+
+  /**
+   * Construct a state with the given client.
+   *
+   * @param client The client for which state is being tracked.
+   * @param owned Is this client owned by the store integration, or was it
+   * provided externally.
+   */
+  constructor(options?: LDRedisOptions, private readonly logger?: LDLogger) {
+    if (options?.client) {
+      this.client = options!.client;
+      this.owned = false;
+    } else if (options?.redisOpts) {
+      this.client = new Redis(options!.redisOpts);
```

Comment:
Unnecessary `!` in these two places:

```suggestion
      this.client = options.client;
      this.owned = false;
    } else if (options?.redisOpts) {
      this.client = new Redis(options.redisOpts);
```